crates/sui-graphql-client/src/lib.rs

```diff
@@ -93,17 +93,30 @@ pub struct Page<T> {
 }
 
 impl<T> Page<T> {
+    /// Return the page information.
     pub fn page_info(&self) -> &PageInfo {
         &self.page_info
     }
 
+    /// Return the data in the page.
     pub fn data(&self) -> &[T] {
         &self.data
     }
 
+    /// Internal function to create a new page with the provided data and page information.
     fn new(page_info: PageInfo, data: Vec<T>) -> Self {
         Self { page_info, data }
     }
+
+    /// Check if the page has data.
+    pub fn is_empty(&self) -> bool {
+        self.data.is_empty()
+    }
+
+    /// Internal function to create a page with no data.
+    fn new_empty() -> Self {
+        Self::new(PageInfo::default(), vec![])
+    }
 }
```

Comment:
You could also `derive(Default)` on `Page` for this.

Reply:
Thanks! I remember trying this and it was complaining about something :D

Reply:
Ah that's a shame, thanks for checking. The issue is that `derive(Default)` introduces a `Default` constraint on all type parameters, even if it's not strictly necessary (e.g. the `Default` impl for `Vec<T>` does not require `Default` on the `T`).